### PR TITLE
VMware: new module: vmware_datastore_cluster

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME
lib/ansible/module_utils/vmware.py
lib/ansible/modules/cloud/vmware/vmware_datastore_cluster.py
test/integration/targets/vmware_datastore_cluster/aliases
test/integration/targets/vmware_datastore_cluster/tasks/main.yml

##### ANSIBLE VERSION

```
2.6devel
```

### CODE REVIEW

File: test/sanity/validate-modules/ignore.txt [hunk 1]
```diff
@@ -1185,134 +1185,87 @@ lib/ansible/modules/cloud/vmware/vca_nat.py E325
...
+lib/ansible/modules/cloud/vmware/vcenter_license.py E326
 lib/ansible/modules/cloud/vmware/vmware_cfg_backup.py E326
-lib/ansible/modules/cloud/vmware/vmware_cluster.py E325
 lib/ansible/modules/cloud/vmware/vmware_cluster.py E326
-lib/ansible/modules/cloud/vmware/vmware_datacenter.py E325
 lib/ansible/modules/cloud/vmware/vmware_datacenter.py E326
-lib/ansible/modules/cloud/vmware/vmware_datastore_facts.py E325
 lib/ansible/modules/cloud/vmware/vmware_datastore_facts.py E326
-lib/ansible/modules/cloud/vmware/vmware_dns_config.py E325
+lib/ansible/modules/cloud/vmware/vmware_datastore_cluster.py E326
```
Comment (mattclay) [inline test/sanity/validate-modules/ignore.txt]: New ignores should not be necessary.

File: lib/ansible/modules/cloud/vmware/vmware_vm_vm_drs_rule.py [hunk 2]
```diff
@@ -46,17 +46,20 @@
     - If set to C(True), the DRS rule will be enabled.
     - Effective only if C(state) is set to C(present).
     default: False
+    type: bool
```
Comment (mattclay) [inline lib/ansible/modules/cloud/vmware/vmware_vm_vm_drs_rule.py]: Changes to files (this and others) unrelated to the new module should be in a separate PR.

**Review by mattclay — CHANGES_REQUESTED** [review]

**Comment by Akasurde** [comment]
@mattclay I have removed all "validate-module" related changes.

**Review by mattclay — APPROVED** [review]

**Comment by Akasurde** [comment]
Closing and re-opening for CI trigger.

**Comment by Akasurde** [comment]
@pdellaert @tchernomax Could you please review this ?

**Comment by Akasurde** [comment]
@pdellaert @tchernomax Could you please review this ?

File: lib/ansible/modules/cloud/vmware/vmware_datastore_cluster.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+      description:
+      - The name of the datacenter.
+      required: True
+    datastore_cluster_name:
+      description:
+      - The name of the datastore cluster.
+      required: True
+    state:
+      description:
+      - If the datacenter should be present or absent
```
Comment (pdellaert) [inline lib/ansible/modules/cloud/vmware/vmware_datastore_cluster.py]: I assume `If the datastore cluster should be present or absent`

**Review by pdellaert — CHANGES_REQUESTED** [review]
Small doc issue

File: lib/ansible/modules/cloud/vmware/vmware_datastore_cluster.py (hunk 3 above)
Comment (Akasurde) [reply lib/ansible/modules/cloud/vmware/vmware_datastore_cluster.py]: OK.

Comment (Akasurde) [reply lib/ansible/modules/cloud/vmware/vmware_datastore_cluster.py]: OK.

**Review by pdellaert — APPROVED** [review]
LGTM

**Comment by Akasurde** [comment]
rebuild_merge
